doc-content/drools-docs/src/main/asciidoc/DecisionEngine/inference-and-truth-maintenance.adoc

```diff
@@ -0,0 +1,271 @@
+[id='inference-and-truth-maintenance_{context}']
+
+= Inference and truth maintenance in the {DECISION_ENGINE}
+
+The basic function of the {DECISION_ENGINE} is to match production facts and data to rules and determine whether and how rules are executed. To ensure that relevant data is applied to the appropriate rules, the {DECISION_ENGINE} makes inferences based on existing knowledge and performs the actions based on the inferred information.
```

Comment:
* Done
* match to and with or both acceptable, but match to has a stronger sense of directionality that the data is what causes rules to execute. With "with", it's not quite as strong of a directinality, more of just an equal pairing. This is definitely a subjective topic that comes up a lot, with differing opinions. In this case, I'd still say match to. Great eye, btw :)
* It's a given, with the audience. This has come up before, and the decision was made that because Facts, objects, etc. are wide-spread concepts in programming and not specific to our product, that it would be difficult where to draw the line of what to define and what not to, and that the target audience in reality knows it, or should.

Reply:
> It's a given, with the audience

ok. Although we can certainly give objects and functions and variables for granted, I'd argue that we should at least mention what rules and facts are; if anything, for reference.